src/librustdoc/core.rs

```diff
@@ -647,10 +643,16 @@ crate fn run_global_ctxt(
 
     ctxt.sess().abort_if_errors();
 
+    let render_options = ctxt.render_options;
+    let mut cache = ctxt.cache;
```

Comment:
Are these locals necessary?

Reply:
They're not strictly necessary, but IMO removing them makes the code harder to read.
```
 diff --git a/src/librustdoc/core.rs b/src/librustdoc/core.rs
index d9669584de3..891ba67f3f5 100644
--- a/src/librustdoc/core.rs
+++ b/src/librustdoc/core.rs
@@ -643,16 +643,16 @@ fn report_deprecated_attr(name: &str, diag: &rustc_errors::Handler) {
 
     ctxt.sess().abort_if_errors();
 
-    let render_options = ctxt.render_options;
-    let mut cache = ctxt.cache;
+    //let render_options = ctxt.render_options;
+    //let mut cache = ctxt.cache;
     krate = tcx.sess.time("create_format_cache", || {
-        cache.populate(krate, tcx, &render_options.extern_html_root_urls, &render_options.output)
+        ctxt.cache.populate(krate, tcx, &ctxt.render_options.extern_html_root_urls, &ctxt.render_options.output)
     });
 
     // The main crate doc comments are always collapsed.
     krate.collapsed = true;
 
-    (krate, render_options, cache)
+    (krate, ctxt.render_options, ctxt.cache)
 }
 
 /// Due to <https://github.com/rust-lang/rust/pull/73566>,
```

Reply:
Okay, I'm okay either way, and I agree that it seems harder to read without the locals.